docs/src/literate-tutorials/dg_heat_equation.jl

```diff
@@ -0,0 +1,382 @@
+# and for each quadrature point we loop over all the (local) shape functions. The *values
+# objects provide values needed to calculate contributions as follows:
+# * `cellvalues`:
+#     * test function value ($\delta u$)
+#     * test and trial functions gradient ($\nabla u$ and $\nabla \delta u$)
+# * `interfacevalues`:
+#     * jump in test and trial functions values
+#     * average of test and trial functions gradients
+#     * trial function value ($u$) for boundary face integrals
+#
+# !!! note "Notation"
+#     Comparing with the brief finite element introduction in [Introduction to FEM](@ref),
+#     the variables `δu`, `∇δu` and `∇u` are actually $\phi_i(\textbf{x}_q)$, $\nabla
+#     \phi_i(\textbf{x}_q)$ and $\nabla \phi_j(\textbf{x}_q)$, i.e. the evaluation of the
+#     trial and test functions in the quadrature point ``\textbf{x}_q``. However, to
+#     underline the strong parallel between the weak form and the implementation, this
+#     example uses the symbols appearing in the weak form.
+
+function assemble_element!(Ke::Matrix, fe::Vector, cellvalues::CellValues)
+    n_basefuncs = getnbasefunctions(cellvalues)
+    ## Reset to 0
+    fill!(Ke, 0)
+    fill!(fe, 0)
+    ## Loop over quadrature points
+    for q_point in 1:getnquadpoints(cellvalues)
+        ## Get the quadrature weight
+        dΩ = getdetJdV(cellvalues, q_point)
+        ## Loop over test shape functions
+        for i in 1:n_basefuncs
+            δu  = shape_value(cellvalues, q_point, i)
+            ∇δu = shape_gradient(cellvalues, q_point, i)
+            ## Add contribution to fe
+            fe[i] += δu * dΩ
+            ## Loop over trial shape functions
+            for j in 1:n_basefuncs
+                ∇u = shape_gradient(cellvalues, q_point, j)
+                ## Add contribution to Ke
+                Ke[i, j] += (∇δu ⋅ ∇u) * dΩ
+            end
+        end
+    end
+    return Ke, fe
+end
+
+function assemble_interface!(Ki::Matrix, iv::InterfaceValues)
+    ## Reset to 0
+    fill!(Ki, 0)
+    ## Loop over quadrature points
+    for q_point in 1:getnquadpoints(iv)
+        ## Get the normal to face A
+        normal = getnormal(iv, q_point)
+        ## Get the quadrature weight
+        dΓ = getdetJdV(iv, q_point)
+        ## Loop over test shape functions
+        for i in 1:getnbasefunctions(iv)
+            ## Multiply the jump by the normal, as the definition used in Ferrite doesn't include the normals.
+            test_jump = shape_value_jump(iv, q_point, i) * normal
+            test_grad_avg = shape_gradient_average(iv, q_point, i)
+            ## Loop over trial shape functions
+            for j in 1:getnbasefunctions(iv)
+                ## Multiply the jump by the normal, as the definition used in Ferrite doesn't include the normals.
+                trial_jump = shape_value_jump(iv, q_point, j) * normal
+                trial_grad_avg = shape_gradient_average(iv, q_point, j)          
```

Comment:
I also might ask for clarification here. Can you expand the 1D heat problem with 2 elements by hand and check if this implementation is really equivalent?

Reply:
Will try to tex my version.